### ISSUE TITLE
Critical errors not displayed aggressively enough for the user, leads to wrong focus on QGC errors

### ISSUE THREAD

**Issue by LorenzMeier** [mavlink/qgroundcontrol#2829]
<img width="1392" alt="screen shot 2016-02-17 at 16 13 58" src="https://cloud.githubusercontent.com/assets/1208119/13113933/87710176-d591-11e5-8fe8-226b2b17b01b.png">

**Comment by LorenzMeier** [comment]
And actually this a usability issue: The real error is shown in a critical message from the system. The critical messages should probably be shown all in the lower message pane:

<img width="1392" alt="screen shot 2016-02-17 at 16 15 10" src="https://cloud.githubusercontent.com/assets/1208119/13113968/b6c69008-d591-11e5-9515-6a9a9a7a833c.png">

**Comment by DonLakeFlyer** [comment]
@dogmaphobic Shouldn'y all critical errors be showing up down there? I have noticed sometimes they don't pop. Not sure why.

**Comment by dogmaphobic** [comment]
I can't find anywhere in the code base where these *critical* messages are ever sent to that message box. There is one handler within Vehicle.cc that receives it but all it does is to send its own signal, which no one listens to:

https://github.com/mavlink/qgroundcontrol/blob/master/src/Vehicle/Vehicle.cc#L928

The only messages that are shown in that lower message area are those that are explicitly called. That is, only fully formatted messages are ever sent there. For example, things like these:

https://github.com/mavlink/qgroundcontrol/blob/master/src/Vehicle/Vehicle.cc#L466-L478

In other words, critical messages coming from vehicles are not currently been sent there at all. I will add code to the signal above (Vehicle.cc#L928) and have the critical messages sent there.

**Comment by dogmaphobic** [comment]
Disregard the above. MainWindowInner has a handler for `formatedMessage` (Vehicle.cc#L928 above). Looking...

**Comment by dogmaphobic** [comment]
Done in #3241. It was a side effect of some changes Don had done, I assume to prevent the message handler from sending messages before the main window was ready to receive them.

**Comment by DonLakeFlyer** [comment]
> I assume to prevent the message handler from sending messages before the main window was ready to receive them.

I vaguely remember something like that
